internal/handlers/scan_sbom.go

```diff
@@ -129,6 +137,9 @@ func (h *ScanSBOMHandler) Handle(message messaging.Message) error {
 	}
 
 	_, err = controllerutil.CreateOrUpdate(ctx, h.k8sClient, vulnerabilityReport, func() error {
+		if vulnerabilityReport.Labels == nil {
+			vulnerabilityReport.Labels = prepareVulnerabilityReportLabels()
+		}
```

Comment:
You can implement the real `mutate()` logic like this, so if the labels are removed somehow, they will be added  back.

```suggestion
		if vulnerabilityReport.Labels == nil {
			vulnerabilityReport.Labels = prepareVulnerabilityReportLabels()
		} else {
		        vulnerabilityReport.Labels["app.kubernetes.io/managed-by"] = LabelValueManagedBy
		        vulnerabilityReport.Labels["app.kubernetes.io/component"] = LabelValueComponent
		        vulnerabilityReport.Labels["app.kubernetes.io/part-of"] = LabelValuePartOf
		}
```

This is probably not something that matters in your scenario, but it's still a good idea to understand how `CreateOrUpdate()` works and how to avoid pitfall like this.  (Think about if you have more than one writer to add different labels to the CR.  )

Also it would be nice to move these logic to something like `mutateVulnerabilityReportLabels()` function.

Reply:
You're right about this. Would it be correct to say that these labels shouldn't be changed by users?
I did consider this approach, but I was concerned that users might modify the labels manually — is that a valid concern?

Reply:
you are right about that, removing is something we should concerned about, let me add it, thanks.